test/TestUtilities/Test.Utility/PackageManagement/TestSolutionManager.cs

```diff
@@ -79,24 +79,27 @@ public TestSolutionManager(string solutionDirectory)
             SolutionDirectory = solutionDirectory;
         }
 
-        public MSBuildNuGetProject AddNewMSBuildProject(string projectName = null, NuGetFramework projectTargetFramework = null, string packagesConfigName = null)
+        public MSBuildNuGetProject AddNewMSBuildProject(string projectName = null, NuGetFramework projectTargetFramework = null, string projectPath = null, bool validateExistingProject = true)
         {
-            var existingProject = Task.Run(async () => await GetNuGetProjectAsync(projectName));
-            existingProject.Wait();
-            if (existingProject.IsCompleted && existingProject.Result != null)
+            if (validateExistingProject)
```

Comment:
Currently the validation just checks the project name.
In practice, projects are stored via 4 different keys, include project id, and project paths. 

The primary key however is full name, which is a project path. 

I think these APIs need refactored, but I don't want to get to that now given that we want to service these changes:

https://github.com/NuGet/Client.Engineering/issues/2855